control-plane-operator/controllers/hostedcontrolplane/creatorupdate_ownerref_enforcer.go

```diff
@@ -2,18 +2,57 @@ package hostedcontrolplane
 
 import (
 	"context"
+	"fmt"
 
+	configv1 "github.com/openshift/api/config/v1"
 	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
 	"github.com/openshift/hypershift/support/config"
 	"github.com/openshift/hypershift/support/upsert"
+	"github.com/openshift/hypershift/support/util"
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func retrieveHostedCLuster(ctx context.Context, c crclient.Client, hclusterNamespacedName string) *hyperv1.HostedCluster {
+	hcluster := &hyperv1.HostedCluster{}
+
+	err := c.Get(ctx, util.ParseNamespacedName(hclusterNamespacedName), hcluster)
+	if err != nil {
+		fmt.Printf("failed to get cluster %q: %s\n", hclusterNamespacedName, err)
+
+		return nil
+	}
+
+	return hcluster
+}
+
 func createOrUpdateWithOwnerRefFactory(upstream upsert.CreateOrUpdateFN) func(*hyperv1.HostedControlPlane) upsert.CreateOrUpdateFN {
 	return func(owner *hyperv1.HostedControlPlane) upsert.CreateOrUpdateFN {
 		ownerRef := config.OwnerRefFrom(owner)
+		hclusterNamespacedName := owner.GetAnnotations()["hypershift.openshift.io/cluster"]
+
+		var hcluster *hyperv1.HostedCluster
+
 		return func(ctx context.Context, c crclient.Client, obj crclient.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error) {
+			isScrapeConfig := false
+
+			switch obj.(type) {
+			case *prometheusoperatorv1.ServiceMonitor:
```

Comment:
Remark that, for now, we are only dealing with `servicemonitor` & `podmonitor` with `apiVersion` set to `@monitoring.coreos.com`.
This code may need to be changed to support `apiVersion` set to `monitoring.rhobs`

Reply:
@Nikokolas3270 we are modifying the groupVersion behind the scenes when we configure the HO for RHOBS monitoring.

Reply:
Ok good to know 👍 , because I was considering adding the following types:
- [servicemonitors.monitoring.coreos.com](https://pkg.go.dev/github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring@v0.62.0-rhobs1/v1#ServiceMonitor)
- [podmonitors.monitoring.coreos.com](https://pkg.go.dev/github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring@v0.62.0-rhobs1/v1#PodMonitor)

Reply:
@csrwng: as we change the `groupVersion` dynamically, do you what is the point of having a `monitoring.coreos.com` and a `monitoring.rhobs` definitions of `servicemonitor` and `podmonitor`?
I mean:
- Do you know if those definition are identical, in which case what was the need to fork (this more a question for OBO)
- If those definitions are not identical, do we only use parts which are identical between both definitions?
- As OBO does not consume `servicemonitor` and `podmonitor` with `groupVersion` set to `monitoring.coreos.com`, is there a plan to change Hypershift code to only support scrape configs with `monitoring.rhobs` as `groupVersion`? That way changing the `groupVersion` will no longer be needed which is a good thing I guess.

Reply:
> Do you know if those definition are identical, in which case what was the need to fork (this more a question for OBO team I guess)

Right now they are, which makes it simple to simply switch the groupVersion when using the API. However, my understanding is that the rhobs version will evolve at a different pace from the monitoring.coreos.com one, potentially picking up changes sooner. If it becomes necessary, we will have to add code in our reconcile loops to use one or the other depending on which API we need to use. However, we will cross that bridge when we get there (if we need to use a feature available in one API and not the other).

> If those definitions are not identical, do we only use parts which are identical between both definitions?

As mentioned above, the plan is to keep using the same type as long as we only need to use parts that are identical. As soon as we need to use features in one API that are not in the other, we'll need to add the conditional logic.

> As OBO does not consume servicemonitor and podmonitor with groupVersion set to monitoring.coreos.com, is there a plan to change Hypershift code to only support scrape configs with monitoring.rhobs as groupVersion? That way changing the groupVersion will no longer be needed which is a good thing I guess.

SD/OBO is not the only use case for the core hypershift code. It is also meant to be used by customers standalone via ACM/MCE in which case we'll want the `monitoring.coreos.com` version

Reply:
Thanks Cesar for the explanations